File: frontend/app/models/pbx-device-binding.js
import DS from 'ember-data';
import pbxDeviceBindingValidations from '../validations/pbxDeviceBinding'

export default DS.Model.extend(pbxDeviceBindingValidations, {
  pbxDeviceId: DS.attr('string'),
  bindingIndex: DS.attr('number'),
  bindingType: DS.attr('string'),
  bindingBehavior: DS.attr('string'),
  bindingDisplay: DS.attr('string'),
  bindingArgument: DS.attr('string'),
  pbxUserId: DS.attr('string'),
  pbxParkingLotId: DS.attr('string'),
  pbxDevice: DS.belongsTo('pbxDevice')
});

File: app/models/pbx_device_binding.rb
class PbxDeviceBinding < RestModel
    include ActiveModel::Validations

    belongs_to :pbx_device

    property :id
    property :pbx_device_id
    property :binding_index
    property :binding_type
    property :binding_behavior
    property :binding_display
    property :binding_argument
    property :pbx_user_id
    property :pbx_parking_lot_id

    validates :pbx_user_id, presence: true, unless: Proc.new { |b| b.binding_behavior == "speeddial" || b.binding_behavior == "park" || b.binding_behavior == "empty" }
    validates :pbx_device_id, presence: true
    
end